### PR TITLE
Add Quake Live

### PR DESCRIPTION
**Pull request by microton79**
### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [X] Does your submission pass tests (server is connectable)?
3. [X] Have you added the server to the main README.md?
4. [X] Have you added a unique README.md for the server you are adding?

### CODE REVIEW

**Comment by DomiiBunn** [comment]
I don't play Quake but just out of interest the last variable 'fps' are you sure it should be fps and not tps(ticks per second) i don't believe server timings will effect client frame times. I might be wrong for Quake

**Comment by Software-Noob** [comment]
You forgot to edit the README in [steamcmd_servers](https://github.com/microton79/eggs/blob/add/quakelive/steamcmd_servers/README.md) also you don't seem to be using the latest parkers Steamcmd install script.

@DomiiBunn Quake uses sv_fps to change how many frames the server runs per second

**Comment by DomiiBunn** [comment]
I have never in my life heard of a server dictating clients fps

**Comment by Software-Noob** [comment]
No one said that it's dictating client FPS, it sets the server FPS the same way for example Rust does.

**Comment by DomiiBunn** [comment]
Than that's not refered as `Frames Per Second` it is going to be `Ticks Per Second` as the server is not drawing frames....

**Comment by Software-Noob** [comment]
Yet, it's still called server FPS in many games, Games can and often do render frames independently from the client. There is an alternative option to alter the tick rate itself. Source games and Rust are some of the popular examples.

**Comment by microton79** [comment]
@Software-Noob Thx, i will add that readme.

**Comment by microton79** [comment]
should i add com_hunkMegs too as a var ? I thought i leave it to the server.cfg but what do you think ?

**Comment by Software-Noob** [comment]
> should i add com_hunkMegs too as a var ? I thought i leave it to the server.cfg but what do you think ?

All up to you and parker will let you know if he approves, personally I only add the basic requirements to run the server as variables and the rest are up to the client/admin to modify or add more variables. You can view examples of other eggs in the repo to see their variable options

**Comment by DomiiBunn** [comment]
> Yet, it's still called server FPS in many games, Games can and often do render frames independently from the client. There is an alternative option to alter the tick rate itself. Source games and Rust are some of the popular examples.

That makes 0 sense to me but I guess that is how it is

**Comment by microton79** [comment]
Yeah. Some Vars shouldnt be able to be changed for the user tho. Like if you offer a 16 slot server you dont want a user to just change it in the configfile. So the question is, is that var something an admin might limit to the user

**Comment by microton79** [comment]
afaik the hunkmegs is mem and mem is something you can limit in ptero anyways.. so i lleave it as it is xD

**Comment by Software-Noob** [comment]
> Yeah. Some Vars shouldnt be able to be changed for the user tho. Like if you offer a 16 slot server you dont want a user to just change it in the configfile. So the question is, is that var something an admin might limit to the user

That would be up to the hosts to configure that themselves, generally parker has been against limiting player slots unless the game requires such variable or option to make it run

**Comment by microton79** [comment]
> > Yeah. Some Vars shouldnt be able to be changed for the user tho. Like if you offer a 16 slot server you dont want a user to just change it in the configfile. So the question is, is that var something an admin might limit to the user
> 
> That would be up to the hosts to configure that themselves, generally parker has been against limiting player slots unless the game requires such variable or option to make it run

Thx, that will make my life easier xD

**Comment by microton79** [comment]
@parkervcp anything i need to change ?

File: steamcmd_servers/quake_live/egg-quake-live.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            "name": "Server Type",
+            "description": "0 = Offline, 1 = LAN, 2 = Internet",
+            "env_variable": "SERVER_TYPE",
+            "default_value": "2",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric|max:2"
+        },
+        {
+            "name": "Max Clients",
```
Comment (parkervcp) [inline steamcmd_servers/quake_live/egg-quake-live.json:55]: As I have called out in many other submitted games I don't like limiting this by default and should be user editable at least if it's required.

Most people use pterodactyl to self-host servers. Making them go to the admin pages is limiting and annoying.

Comment (parkervcp) [reply steamcmd_servers/quake_live/egg-quake-live.json:55]: Beyond the max players I think the rest looks fine.

Comment (microton79) [reply steamcmd_servers/quake_live/egg-quake-live.json]: done

**Comment by microton79** [comment]
@parkervcp am i still missing something?

**Comment by parkervcp** [comment]
I haven't looked again recently.

**Pull request merged**